x/dispensation/keeper/mint_controller.go

```diff
@@ -0,0 +1,35 @@
+package keeper
+
+import (
+	clptypes "github.com/Sifchain/sifnode/x/clp/types"
+	"github.com/Sifchain/sifnode/x/dispensation/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) SetMintController(ctx sdk.Context, mintController types.MintController) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.MintControllerPrefix, k.cdc.MustMarshal(&mintController))
+}
+
+func (k Keeper) GetMintController(ctx sdk.Context) types.MintController {
+	controller := types.MintController{}
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.MintControllerPrefix)
+	k.cdc.MustUnmarshal(bz, &controller)
+	return controller
+}
+
+func (k Keeper) AddMintAmount(ctx sdk.Context, mintedCoin sdk.Coin) {
+	controller := k.GetMintController(ctx)
+	controller.TotalCounter = controller.TotalCounter.Add(mintedCoin)
+	k.SetMintController(ctx, controller)
+}
+
+func (k Keeper) TokensCanBeMinted(ctx sdk.Context) bool {
```

Comment:
Where is this called from currently?

Reply:
being called in the block beginner according to Tanmay

Reply:
http://github.com/Sifchain/sifnode/blob/650839ad5a10c8164e68e9cb066d2abc65866d4e/x/dispensation/abci.go#L12-L14